### ISSUE TITLE
`prefer_type_checking` and optional types violation

### ISSUE THREAD

**Issue by cyberowl** [realm/SwiftLint#5802]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

The following code causes false positive `prefer_type_checking` violation:

```swift
let value: String? = nil
print(value as Any? != nil) // -> false
```

Let's try fix it:
```
swiftlint . --fix
```

Now we got this:
```swift
let value: String? = nil
// xcodebuild warning: `'is' test is always true`
print(value is Any?) // -> now true, that's not expected
```

### Environment
- SwiftLint version: 0.57.0
- Xcode version: 16.0 (16A242d)
- Installation method used: Homebrew
- Configuration file: None

**Comment by mildm8nnered** [comment]
At first I thought we should just ignore `SomeType?` completely, but actually I wonder if we should be triggering on `as` at all, as opposed to `as?`?

That would fix this case.

`someVar as? SomeType? != nil` would still get caught, but we could ignore `SomeType?` as well

**Comment by mildm8nnered** [comment]
So I think this case will also be problematic

```
let x: String? = nil
print(x as String? != nil) // false
print(x is String?) // true
```

So I think we want to exclude the OptionalTypeSyntax cases as well
